feat: Add docs for new native features (#1212)
getsentry/sentry-docs: src/collections/_documentation/workflow/debug-files.md

```

`sentry-cli` can detect these differences during the upload process and
associates the same identifier to both files. However, this requires that both
files are uploaded in the same invocation of the upload command. Otherwise, the
identifiers diverge and Sentry might not be able to resolve the correct file
for symbolication.

### ProGuard UUIDs

Unlike other debug information files, ProGuard files do not have an intrinsic
unique identifier. Sentry CLI assigns them a SHA1 UUID based on the checksum of
the file. You can use `sentry-cli difutil check` on a ProGuard file to see the
generated UUID.

If you need to generate the UUID yourself, you can do so with the following
algorithm (Python code for reference):

```python
import uuid

NAMESPACE = uuid.uuid5(uuid.NAMESPACE_DNS, "guardsquare.com")

def get_proguard_uuid(filename):
    with open(filename, 'rb') as f:
        return uuid.uuid5(NAMESPACE, f.read())
```

## Uploading Files

The most straight-forward way to provide Sentry with debug information file is
to upload them using `sentry-cli`. Depending on your workflow, you may want to
upload as part of your build pipeline or when deploying and publishing your
application:

> [Visualization release process]

Author: Correct.

  file contents, commonly used for source file lookups. Sentry does not support
  this lookup method.
  
  Examples:
   - `wkernel32.pdb/ff9f9f7841db88f0cdeda9e1e9bff3b5A/wkernel32.pdb`
   - `kernel32.dll/590285e9e0000/kernel32.dll`
   - `libc-2.23.so/elf-buildid-b5381a457906d279073822a5ceb24c4bfef94ddb/libc-2.23.so`
   - `_.debug/elf-buildid-sym-b5381a457906d279073822a5ceb24c4bfef94ddb/_.debug`
   - `CoreFoundation/mach-uuid-36385a3a60d332dbbf55c6d8931a7aa6/CoreFoundation`
   - `_.dwarf/mach-uuid-sym-36385a3a60d332dbbf55c6d8931a7aa6/_.dwarf`

**SymStore**

: Path: `&lt;FileName&gt;/&lt;SIGNATURE&gt;&lt;AGE&gt;/&lt;FileName&gt;`
  
  The public symbol server provided by Microsoft used to only host PDBs for the
  Windows platform. These use a _signature-age_ debug identifier in addition to
  the file name to locate symbols. File paths are identical to SSQP, except for
  the default casing rules:
  
   - Filenames are **as given**
   - The signature and age of a PDB identifier are **uppercase**.
   - The timestamp of a PE identifier is **uppercase**, but the size is
     **lowercase**.

  Since the original Microsoft Symbol Server did not serve ELF or MachO files,

>   we do not recommend to use this convention for these types. However, Sentry

Reviewer: ```suggestion
  we do not recommend using this convention for these types. However, Sentry
```

  will support the SSQP conventions with adapted casing rules when this layout

>   selected.

Reviewer: ```suggestion
  is selected.
```

[ssqp key conventions]: https://github.com/dotnet/symstore/blob/master/docs/specs/SSQP_Key_Conventions.md
[file mapped uuid directories]: http://lldb.llvm.org/use/symbols.html#file-mapped-uuid-directories
[build id method]: https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html

### Compression of Debug Files

Sentry supports the following compression methods when downloading debug
information files from external sources: Gzip, zlib (both with and
without header), Zstandard, and Cabinet (CAB).

The convention on Microsoft's Symbol Server protocol is to store such files with
the last character of the file extension replaced with `_`. A full example would
be: `KERNEL32.dll/590285E9e0000/KERNEL32.dl_`. This is not required on your own
repositories, as Sentry detects compression on all paths.

## Source Context

If Sentry has access to application source code, it can show snippets of code
around the location of stack frames. Certain SDKs can resolve this source
context automatically, such as the Python SDK, because they have access to
unobfuscated source code at runtime.

To get source context for native applications, source code needs to be uploaded

> alongside the debug information files. The recommended way to do this is using

Reviewer: ```suggestion
alongside the debug information files. The recommended way to do this is by using
```

`sentry-cli`. See [Creating Source Bundles]({%- link _documentation/cli/dif.md
-%}#creating-source-bundles) for more information.

Source bundles show up as regular debug files on the _Debug Files_ settings

> page. They are designated as "source bundle" and feature a `sources` tag. In

Reviewer: ```suggestion
page. They are designated as "source bundle" and feature a `sources` tag.
```

> order to match them with crash reports, they carry the same debug file as the

Reviewer: ```suggestion
To match them with crash reports, they carry the same debug file as the
```
